agent/command/shell.go

```diff
@@ -82,6 +82,13 @@ func (*shellExec) Name() string { return evergreen.ShellExecCommandName }
 
 // ParseParams reads in the command's parameters.
 func (c *shellExec) ParseParams(params map[string]interface{}) error {
+	if params == nil {
+		return errors.New("params cannot be nil")
+	}
+	if params["script"] == nil {
+		return errors.Errorf("command '%s' specified without a script", evergreen.ShellExecCommandName)
```

Comment:
I think this can just return the error "script cannot be nil" since https://github.com/evergreen-ci/evergreen/blob/d90ee9d7f77d5d29f58f8231b9e5217206091f07/validator/project_validator.go#L943 handles the command name.